lldb/tools/lldb-rpc/LLDBRPCGeneration.cmake

```diff
@@ -78,3 +78,7 @@ add_custom_target(lldb-rpc-generate-sources
   lldb-sbapi-dwarf-enums)
 
 add_dependencies(lldb-rpc-generate-sources clang-resource-headers)
+
+# Add a compile definition to liblldb in order to create a macro for
+# a Clang attribute for lldb-rpc-gen.
+target_compile_definitions(liblldb PUBLIC LLDB_RPC_GEN)
```

Comment:
I don't think this does what you think it does. This tells CMake that it should compile all files associated with liblldb with `-DLLDB_RPC_GEN`. However, `LLDB_RPC_GEN` should only be defined when `lldb-rpc-gen` is ingesting the SBAPI headers.

What you want instead is to add an extra argument to the `lldb-rpc-gen` invocation so it passes `-DLLDB_RPC_GEN` to its clang tool.